idb/postgres/internal/writer/writer.go

```diff
@@ -290,6 +299,30 @@ func writeAccountDeltas(round basics.Round, accountDeltas *ledgercore.AccountDel
 			writeAppResource(round, &appResources[i], batch)
 		}
 	}
+
+}
+
+func writeBoxMods(kvMods map[string]*string, batch *pgx.Batch) error {
+	// Insert/Update/Delete ON `app_box`
+	// kvMods can in theory support more general storage types than app boxes.
+	// However, here we assume that all the provided kvMods represent app boxes.
+
+	fmt.Printf("writeAccountDeltas - app_box portion for kvMods of len=%d", len(kvMods))
+
+	for key, value := range kvMods {
+		app, name, err := logic.SplitBoxKey(key)
```

Comment:
Upon further reflection, I'd like to keep this immediate exit as it is. If in the future, a new kind of kvstore element is introduced to `kvMods` the indexer will get stuck ingesting this round. I verified by returning an error intentionally inside the for-loop. When running my simulated migration, I got stuck at round 9 with the following error:

```
err: AddBlock() err on boxes: writeBoxMods() - this is a temporary error!!!!","level":"error","msg":"block 9 import failed","time":"2022-07-01T12:35:32-05:00"
```

In other words, by leaving this as is we're insisting that we know what to do with non-boxes _before_ we enter this method, or modify it to handle correctly. @winder what do you think?  In terms of conventions for this repo, would you rather fail in the future if a new kind of kvstore element is introduced, or just skip silently?

Reply:
I prefer failing in the future if a new kind of kvstore element is introduced.

Reply:
I agree with Shiqi here. We need to fail whenever we come across something unknown because we only process blocks one time. Migrations to repair the database after the fact can be tricky, so we avoid processing partial blocks as a policy.

Reply:
There are things that happen on the block chain that we don't store a permanent record of because we decided not to. App state, for example.  If we ever wanted to add the feature "What was the app state at round R", we would have to re-index.  Similarly, if we add some random new feature that uses the kvstore, we may or may not wish to support indexing it. I don't think the indexer should crash in such a situation, it should keep going. That _forces_  us to do work in indexer even if we have decided we don't care about this new unknown thing.

Reply:
I yield to you guys - it's your code.